src/commonMain/kotlin/org/treeWare/model/operator/Difference.kt

```diff
@@ -0,0 +1,407 @@
+            updateEnumeration.copyValueFrom(lastEnumeration)
+            //println("Updated enumeration found")
+        }
+        return TraversalAction.CONTINUE
+    }
+
+    override fun visitAssociation(leaderValueList: List<AssociationModel?>): TraversalAction {
+        val lastAssociation = leaderValueList.lastNotNullOf { it }
+
+        /**=====Check if it's absent in the original=====**/
+        if(leaderValueList.first() == null) { //if the original doesn't have the entity:
+            stepsFromAdd = 0 //everything above this node should be included
+            val addParent = addStack.first()
+            val addAssociation = newChildValue(addParent) as MutableAssociationModel
+            copy(lastAssociation, addAssociation)
+            //println("New association found")
+        }
+
+        /**=====Check if it's absent in the new version=====**/
+        else if(leaderValueList.last() == null) {//if the new version doesn't have the entity:
+            stepsFromDelete = 0 //everything above this node should be included
+            val deleteParent = deleteStack.first()
+            val deleteAssociation = newChildValue(deleteParent) as MutableAssociationModel
+            copy(lastAssociation, deleteAssociation)
+            //println("Deleted association found")
+        }
+
+        /**=====Check if it's been updated =====**/
+        else if(leaderValueList.last() != leaderValueList.first()) {//if old and new aren't the same:
+            stepsFromUpdate = 0 //everything above this node should be included
+            val updateParent = updateStack.first()
+            val updateAssociation = newChildValue(updateParent) as MutableAssociationModel
+            copy(lastAssociation, updateAssociation)
+            //println("Updated association found")
+        }
+        return TraversalAction.CONTINUE
+    }
+
+    // Helpers
+
+    private fun visitField(leaderFieldList: List<FieldModel?>): TraversalAction {
+        val lastLeaderField = leaderFieldList.lastNotNullOf { it }
+        val lastLeaderFieldName = getFieldName(lastLeaderField)
+
+        val addParent = addStack.first() as MutableBaseEntityModel
+        val addField = newDisconnectedField(addParent,lastLeaderFieldName)
+        addStack.addFirst(addField)
+        stepsFromAdd ++
+
+        val deleteParent = deleteStack.first() as MutableBaseEntityModel
+        val deleteField = newDisconnectedField(deleteParent,lastLeaderFieldName)
+        deleteStack.addFirst(deleteField)
+        stepsFromDelete ++
+
+        val updateParent = updateStack.first() as MutableBaseEntityModel
+        val updateField = newDisconnectedField(updateParent,lastLeaderFieldName)
+        updateStack.addFirst(updateField)
+        stepsFromUpdate ++
+
+        /**=====Check if it's absent in the original=====**/
+        if(leaderFieldList.first() == null) { //if the original doesn't have the entity:
```

Comment:
copy/paste error? s/entity/field/

Applies to the comment in the next if-block as well.